oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/osgi/OakSolrConfigurationProviderService.java

```diff
@@ -135,25 +176,25 @@ public class OakSolrConfigurationProviderService implements OakSolrConfiguration
     private OakSolrConfiguration oakSolrConfiguration;
 
     @Activate
-    protected void activate(ComponentContext componentContext) throws Exception {
-        pathChildrenFieldName = String.valueOf(componentContext.getProperties().get(PATH_CHILDREN_FIELD));
-        pathParentFieldName = String.valueOf(componentContext.getProperties().get(PATH_PARENT_FIELD));
-        pathExactFieldName = String.valueOf(componentContext.getProperties().get(PATH_EXACT_FIELD));
-        collapsedPathField= String.valueOf(componentContext.getProperties().get(COLLAPSED_PATH_FIELD));
-        pathDescendantsFieldName = String.valueOf(componentContext.getProperties().get(PATH_DESCENDANTS_FIELD));
-        catchAllField = String.valueOf(componentContext.getProperties().get(CATCH_ALL_FIELD));
-        depthField = String.valueOf(componentContext.getProperties().get(PATH_DEPTH_FIELD));
-        rows = Integer.parseInt(String.valueOf(componentContext.getProperties().get(ROWS)));
-        commitPolicy = OakSolrConfiguration.CommitPolicy.valueOf(String.valueOf(componentContext.getProperties().get(COMMIT_POLICY)));
-        useForPathRestrictions = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(PATH_RESTRICTIONS)));
-        useForPropertyRestrictions = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(PROPERTY_RESTRICTIONS)));
-        useForPrimaryTypes = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(PRIMARY_TYPES_RESTRICTIONS)));
-        typeMappings = PropertiesUtil.toStringArray(componentContext.getProperties().get(TYPE_MAPPINGS));
-        ignoredProperties = PropertiesUtil.toStringArray(componentContext.getProperties().get(IGNORED_PROPERTIES));
-        usedProperties = PropertiesUtil.toStringArray(componentContext.getProperties().get(USED_PROPERTIES));
-        propertyMappings = PropertiesUtil.toStringArray(componentContext.getProperties().get(PROPERTY_MAPPINGS));
-        collapseJcrContentNodes = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(COLLAPSE_JCR_CONTENT_NODES)));
-        collapseJcrContentParents = Boolean.valueOf(String.valueOf(componentContext.getProperties().get(COLLAPSE_JCR_CONTENT_PARENTS)));
+    protected void activate(Configuration configuration) {
+        pathChildrenFieldName = configuration.path_child_field();
```

Comment:
I would remove this indirection, just store the object

Reply:
Imho hat would need a new Jira issue.